### ISSUE TITLE
building on macOS

### ISSUE THREAD

**Issue by bucanero** [HackerSmacker/CSave#1]
not really an issue, but I wanted to share how to build your Borderland tools on macOS:

- install protobuf-c: 
   ```sh
   brew install protobuf-c
   ```
- modify the Makefile:
   ```
   # Link flags. 
   LDFLAGS = -lprotobuf-c -L/usr/local/lib -L/usr/lib -L/opt/homebrew/opt/protobuf-c/lib
   CFLAGS = -I. -O0 -I/opt/homebrew/opt/protobuf-c/include
   ```
- build with `make`

thanks for sharing your tools 👍 
once I fully understand the save structure and your code, I'll try to add the PS4 save decryption to my ps4 homebrew save manager [Apollo](https://github.com/bucanero/apollo-ps4)

**Comment by HackerSmacker** [comment]
Yeah, it works on just about all UNIX flavors (albeit, compiling protobuf-c can be somewhat difficult on some systems). Seems like a lot of people use this program to work with console saves -- I haven't tested it myself (I don't have much interest in it), but I've read that it works fine. Now, as for the save decryption for BL3 saves, that's some BL3-specific thing, it doesn't exist anywhere else (that I could find). 

I should really re-write this thing, it was a quick-n-dirty program I wrote to maintain a catalog of all of my BL3 saves (of which I have probably two hundred or so)!

Anyways, good luck!

**Comment by bucanero** [comment]
I was testing with my ps4 saves and the encryption/decryption works fine. I didn't test the other tools, but I think they'd work fine with the ps4 saves too.

Just for reference, here's my simplified code that only decrypt/encrypt the custom protection on PS4 saves:
https://github.com/bucanero/ps3-save-decrypters/tree/master/ps4-borderlands3-decrypter

The encryption is based on a simple XOR process, but as you mentioned this looks like a game-specific custom thing and not a standard encryption algorithm like AES, DES, etc.
